### PR TITLE
Make is_workflow_required an explicit section

### PR DESCRIPTION
**Pull request by robnewman**
# Description
Make `is_workflow_required` an explicit section with a clear example of the related `config.yml` file

### CODE REVIEW

File: docs/advanced-features/workflows.md [hunk 1]
```diff
@@ -185,9 +185,19 @@ quilt3.Package().set_meta({'answer': 42}).push(
 # Package test/package@6331508 pushed to s3://quilt-sergey-dev-metadata
 ```
 
-If you wish for your users to be able to skip workflows altogether, you can make
-workflow validation optional with `is_workflow_required: false` in your `config.yml`,
-and specify `workflow=None` in the API:
+## Bypassing workflow validation and setting a default workflow
```
Comment (QuiltSimon) [inline docs/advanced-features/workflows.md:188]: My intended goal in requesting the update to this document was to help users understand how to require workflows. Though the inverse is also helpful, I'd like to help users understand how to enforce workflows using `is_workflow_required: True`, and how to best utilize this. 

If `is_workflow_required` is not explicitly called out, what is the default state?

File: docs/advanced-features/workflows.md [hunk 2]
```diff
@@ -200,12 +210,32 @@ quilt3.Package().push(
 # Package test/package@06b2815 pushed to s3://quilt-sergey-dev-metadata
```
Comment (QuiltSimon) [inline docs/advanced-features/workflows.md:210]: any chance we can make the S3 bucket name more end-user sanitized? Nothing against Sergey by any means, but I'd like this to be something along the lines of s3://quilt-docs-metadata

File: docs/advanced-features/workflows.md [hunk 3]
```diff
@@ -200,12 +210,32 @@ quilt3.Package().push(
...
+  catalog: "1"
+default_workflow: "experiment"
+is_workflow_required: False
+workflows:
+  experiment:
+    name: Experiment
+    metadata_schema: experiment-universal
+schemas:
+  experiment-universal:
+    url: s3://quilt-sergey-dev-metadata/.quilt/workflows/schemas/experiment-universal.json
```
Comment (QuiltSimon) [inline docs/advanced-features/workflows.md:228]: Echoing the previous comment about generalizing the S3 bucket name

**Review by QuiltSimon — CHANGES_REQUESTED** [review]

File: docs/advanced-features/workflows.md [hunk 4]
```diff
@@ -185,9 +185,19 @@ quilt3.Package().set_meta({'answer': 42}).push(
...
+false` at the top-level in your `config.yml` file:
+
+```yaml
+version:
+  base: "1"
+  catalog: "1"
+is_workflow_required: False
+```
+
+Then specify `workflow=None` in any `Package.push()` API calls:
```
Comment (akarve) [inline docs/advanced-features/workflows.md:200]:
```suggestion
Now your users can specify `workflow=None` (or `--workflow ''`) when they push packages.
```

File: docs/advanced-features/workflows.md (hunk 1 above)
Comment (akarve) [reply docs/advanced-features/workflows.md:188]: By default workflows are required. We should state that explicitly.

**Review by akarve — DISMISSED** [review]
Pre-emptively approving modulo my comments

File: docs/advanced-features/workflows.md (hunk 1 above)
Comment (robnewman) [reply docs/advanced-features/workflows.md:190]: Fixed!

File: docs/advanced-features/workflows.md (hunk 2 above)
Comment (robnewman) [reply docs/advanced-features/workflows.md]: Removed all occurrences of Sergey's name

File: docs/advanced-features/workflows.md (hunk 4 above)
Comment (robnewman) [reply docs/advanced-features/workflows.md]: Fixed - thanks

File: docs/advanced-features/workflows.md (hunk 3 above)
Comment (robnewman) [reply docs/advanced-features/workflows.md]: This is now fixed throughout

**Review by QuiltSimon — APPROVED** [review]
This looks like it solves for my asks. Good work!

**Pull request merged**